src/compiler/types.ts

```diff
@@ -2161,7 +2161,7 @@ namespace ts {
 
     export interface JSDocAugmentsTag extends JSDocTag {
         kind: SyntaxKind.JSDocAugmentsTag;
-        typeExpression: JSDocTypeExpression;
+        class: ExpressionWithTypeArguments & { expression: Identifier | PropertyAccessEntityNameExpression };

```

Comment:
don't you want a subclass of ExpressionWithTypeArguments that has the override `expression: Identifier | PropertyAccessEntityNameExpression` ? Seems like intersection would be too permissive.

Reply:
How is intersection permissive where `extends` isn't? Could you give an example?

Reply:
I forgot that `Identifier` is assignable to `Expression & Identifier` but (for example) `BinaryExpression` isn't. Never mind.